WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginDetailActivity.java

```diff
@@ -0,0 +1,178 @@
+        }
+
+        mPlugin = mPluginStore.getSitePluginByName(mSite, pluginName);
+
+        if (mSite == null) {
+            ToastUtils.showToast(this, R.string.blog_not_found, Duration.SHORT);
+            finish();
+            return;
+        }
+
+        if (mPlugin == null) {
+            ToastUtils.showToast(this, R.string.plugin_not_found, Duration.SHORT);
+            finish();
+            return;
+        }
+
+        Toolbar toolbar = findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        ActionBar actionBar = getSupportActionBar();
+        if (actionBar != null) {
+            actionBar.setHomeButtonEnabled(true);
+            actionBar.setDisplayHomeAsUpEnabled(true);
+            actionBar.setElevation(0);
+        }
+
+        setupViews();
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(final MenuItem item) {
+        if (item.getItemId() == android.R.id.home) {
+            onBackPressed();
+            return true;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putSerializable(WordPress.SITE, mSite);
+        outState.putString(KEY_PLUGIN_NAME, mPlugin.getName());
+    }
+
+    private void setupViews() {
+        mSwitchActive = findViewById(R.id.plugin_state_active);
+        mSwitchAutoupdates = findViewById(R.id.plugin_state_autoupdates);
+
+        mSwitchActive.setOnCheckedChangeListener(new OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(CompoundButton compoundButton, boolean b) {
+                if (compoundButton.isPressed()) {
+                    mPlugin.setIsActive(b);
+                    dispatchUpdateAction();
+                }
+            }
+        });
+
+        mSwitchAutoupdates.setOnCheckedChangeListener(new OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(CompoundButton compoundButton, boolean b) {
+                if (compoundButton.isPressed()) {
+                    mPlugin.setIsAutoUpdateEnabled(b);
+                    dispatchUpdateAction();
+                }
+            }
+        });
+
+        findViewById(R.id.plugin_wp_org_page).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                ActivityLauncher.openUrlExternal(PluginDetailActivity.this, getWpOrgPluginUrl());
+            }
+        });
+
+        findViewById(R.id.plugin_home_page).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                ActivityLauncher.openUrlExternal(PluginDetailActivity.this, mPlugin.getPluginUrl());
+            }
+        });
+
+        refreshStates();
+    }
+
+    private void refreshStates() {
+        mSwitchActive.setChecked(mPlugin.isActive());
```

Comment:
Might be worth adding an `isFinishing()` check before updating the views. WDYT?

Reply:
Good call! Fixed in f39bc62ac2229e511e7c785d34d417158910db0c